FordLabs/retroquest
# Move error messages from status 'reason' to response body as per HTTP…

## Overview
This PR changes the way RetroQuest's API reports errors.

*Previously*, RetroQuest API placed error details in the `reason` field of the HTTP status object transmitted as part of the response to the failed request.  This is off-spec behavior, but more importantly: many browsers and other tools *discard status reasons*, as they are redundant - they should always correspond to the status code.  404 is always "Not Found", 409 is always "Conflict", etc.  These are not meant to change per application!

*After this PR is applied*, RetroQuest will put error messages in the body of the response, under the `reason` field.

## Testing Instructions

* Run the app
* Open your browser's inspector
* Behave badly, i.e. create the same team twice, try to login incorrectly, try to reset password for a nonexistent team, etc.
* Note that the same error messages that used to appear in the UI still do, this should be unchanged
* Note that each failed request in the Inspector now comes with a response body including error message
